src/seedu/addressbook/Main.java

```diff
@@ -79,15 +83,27 @@ private void exit() {
     private void runCommandLoopUntilExitCommand() {
         Command command;
         do {
-            String userCommandText = ui.getUserCommand();
-            command = new Parser().parseCommand(userCommandText);
-            CommandResult result = executeCommand(command);
-            recordResult(result);
-            ui.showResultToUser(result);
+            try {
+                String userCommandText = ui.getUserCommand();
+                command = new Parser().parseCommand(userCommandText);
+                isStorageFileExist();
```

Comment:
The exception should be raised from the methods that you are calling. The way you did with exception is more like `if (isThereProblem(...)) then manageProblem(...)`, which is just a regularly checking. 
Please think carefully about handling exceptions (possibly by looking some examples about handing exception, and thinking the differences of implementation between yours and theirs), and discuss with your classmates or tutor if necessary.